BEIMA.Backend.Test/DeviceFunctions/AddDeviceTest.cs

```diff
@@ -24,13 +30,41 @@ public async Task NoDevice_AddDevice_ReturnsValidId()
                   .Verifiable();
             MongoDefinition.MongoInstance = mockDb.Object;
 
+            // Setup storage provider.
+            var services = new ServiceCollection();
+            services.AddSingleton<IStorageProvider, AzureStorageProvider>();
+            var serviceProivder = services.BuildServiceProvider();
+            var storageProvider = serviceProivder.GetRequiredService<IStorageProvider>();
+
             // Create request
-            var body = TestData._testDevice;
-            var request = CreateHttpRequest(RequestMethod.POST, body: body);
+            var data = new AddDeviceRequest()
```

Comment:
Let's put these objects to TestData.cs like we had with the json TestData, I think it just makes the tests a little more readable and allows us to reuse the data for multiple tests.